src/DocumentFormat.OpenXml.Framework/OpenXmlElement.cs

```diff
@@ -1330,6 +1330,21 @@ public bool IsBefore(OpenXmlElement element)
             return GetOrder(this, element) == ElementOrder.Before;
         }
 
+        /// <summary>
+        /// Determines if the specified element is a valid child of the current element.
+        /// </summary>
+        /// <param name="element">The element to check.</param>
+        /// <returns>True if the specified element is a valid child; otherwise, false.</returns>
+        public bool IsValidChild(OpenXmlElement element)
+        {
+            if (element is null)
+            {
+                throw new ArgumentNullException(nameof(element));
+            }
+
+            return Metadata.Children.Elements.Any(el => el.Type.Name.Equals(element.Metadata.Type.Name));
```

Comment:
Wouldn't this only apply on OpenXmlCompositeElement? Should this be virtual and default to false and overriden on that type with this implementation?

Reply:
That's true, updated in latest commit